base/namedtuple.jl

```diff
@@ -286,7 +286,8 @@ function merge(a::NamedTuple, itr)
     names = Symbol[]
     vals = Any[]
     inds = IdDict{Symbol,Int}()
-    for (k::Symbol, v) in itr
```

Comment:
Does the type declaration really have no effect here? If so that seems like a bug.

Reply:
Or I guess the `convert` call is the problem?

Reply:
Yep. Of course if #39419 is doable, then it wouldn't be.